fuel-core/src/schema/block.rs

```diff
@@ -436,9 +436,11 @@ impl BlockMutation {
                 vec![],
             );
 
-            let ExecutionResult { block, .. } =
-                executor.execute(ExecutionBlock::Production(block))?;
-            seal_block(&config.consensus_key, &block, &mut executor.database)?;
+            let (ExecutionResult { block, .. }, mut db_transaction) = executor
+                .execute_without_commit(ExecutionBlock::Production(block))?
+                .into();
+            seal_block(&config.consensus_key, &block, db_transaction.database_mut())?;
```

Comment:
I wonder if there's a way we could reuse the CM to handle the production / sealing of the block, that way we don't need to be concerned with ensuring the blocks are committed with the expected sealed metadata here 🤔

Reply:
Yep, I will add TODO for that